oak-run/src/main/java/org/apache/jackrabbit/oak/run/DataStoreCommand.java

```diff
@@ -284,12 +407,57 @@ private void traverseChildren(NodeState state, String path, ReferenceCollector c
                     for (String node: nodeList) {
                         state = state.getChildNode(node);
                     }
-                    traverseChildren(state, path, collector);
+
+                    if (inclusionRegex == null || inclusionRegex.length == 0) {
+                        traverseChildren(state, path, collector);
+                    } else {
+                        for (String regex : inclusionRegex) {
+                            Map<NodeState, String> inclusionMap = new HashMap<NodeState, String>();
+                            getInclusionListFromRegex(state, path, regex, inclusionMap);
+                            if (inclusionMap.size() == 0 ) {
+                                System.out.println("No Valid paths found for traversal, " +
+                                        "for the inclusion Regex " + regex + " under the path " + path);
```

Comment:
Nit: a bit inconsistent uppercase / lowercase... I would use "No valid path ... for inclusion regex ..."

Reply:
Done